### PR TITLE
refactor Integration tests

### PR DESCRIPTION
**Pull request by tardyp**

### CODE REVIEW

File: master/buildbot/master.py [hunk 1]
```diff
@@ -139,6 +139,9 @@ def __init__(self, basedir, configFileName=None, umask=None, reactor=None, confi
         self.name = self.name.decode('ascii', 'replace')
         self.masterid = None
 
+    def stopReactor(self):
+        self.reactor.stop()
```
Comment (tardyp) [inline master/buildbot/master.py]: patching reactor.stop is not enough anymore with recent twisted, as it is repatched in _wait(), the function, which waits for the test's deferred.

**Comment by rutsky** [comment]
Lint fails: 

```
23.09s$ [ $TESTS != lint ] || make flake8

make -C master flake8

make[1]: Entering directory `/home/travis/build/buildbot/buildbot/master'

flake8 --config=../common/flake8rc buildbot

buildbot/test/util/db.py:52:1: E302 expected 2 blank lines, found 1

class RealDatabaseMixin(object):

^

buildbot/test/util/integration.py:20:1: F401 'mock' imported but unused

import mock

^

1       E302 expected 2 blank lines, found 1

1       F401 'mock' imported but unused

2

make[1]: *** [flake8] Error 1

make[1]: Leaving directory `/home/travis/build/buildbot/buildbot/master'

make: *** [flake8] Error 2
```

Plus CI fails on some configurations (PostreSQL, non-temporary SQLite DB).

**Comment by tardyp** [comment]
needs work to reset the db

**Comment by tardyp** [comment]
I am not convinced anymore of that refactoring.

**Pull request closed without merging**